darwin/dataset/utils.py

```diff
@@ -0,0 +1,291 @@
+import json
+import os
+from collections import defaultdict
+from pathlib import Path
+from typing import List, Optional, Iterable
+from sklearn.model_selection import train_test_split
+
+import numpy as np
+
+def extract_classes(annotation_files: List, annotation_type: str):
+    """
+    Given a the GT as json files extracts all classes and an maps images index to classes
+
+    Parameters
+    ----------
+    annotation_files: list
+        List of json files with the GT information of each image
+    annotation_type : str
+        Type of annotation to use to extract the Gt information
+
+    Returns
+    -------
+    classes: dict
+    Dictionary where keys are the classes found in the GT and values
+    are a list of file numbers which contain it
+    idx_to_classes: dict
+    Dictionary where keys are image indices and values are all classes
+    contained in that image
+    """
+    classes = defaultdict(set)
+    indices_to_classes = defaultdict(set)
+    for i, file_name in enumerate(annotation_files):
+        with open(file_name) as f:
+            for annotation in json.load(f)["annotations"]:
+                if annotation_type not in annotation:
+                    continue
+                class_name = annotation["name"]
+                indices_to_classes[i].add(class_name)
+                classes[class_name].add(i)
+    return classes, indices_to_classes
+
+def make_class_list(
+        file_name: str,
+        annotation_files: List,
+        lists_path: Path,
+        annotation_type: str,
+        force_resplit: Optional[bool] = False,
+        add_background: Optional[bool] = False
+):
+    """
+    Support function to extract classes and save the output to file
+
+    Parameters
+    ----------
+    file_name : str
+        Name of the file where to store the results
+    annotation_files : list
+        List of json files with the GT information of each image
+    lists_path : Path
+        Path to the lists folder
+    annotation_type : str
+        Type of annotations to use, e.g. 'tag' or 'polygon'
+    force_resplit : bool
+        Force the creation of the output file, should the list already exist
+    add_background : bool
+        Add the '__background__' class to the list of classes
+
+    Returns
+    -------
+    idx_to_classes: dict
+    Dictionary where keys are image indices and values are all classes
+    contained in that image
+    """
+    fname = lists_path / file_name
```

Comment:
Very minor, but we should use the same naming convention throughout the codebase. Is it `filename`, `file_name` or `fname`? Either way, let's choose one and go for it!

P.S.: my vote would go to `filename` but you can totally ignore that 😊

Reply:
Done (`filename` chosen)